indexing-service/src/main/java/io/druid/indexing/worker/executor/ExecutorLifecycle.java

```diff
@@ -91,30 +111,6 @@ public void start()
       throw Throwables.propagate(e);
     }
 
-    // Spawn monitor thread to keep a watch on parent's stdin
```

Comment:
I think an easy way to make sure all the tasks are dead when killing a middleManager would be useful. Sometimes you really do want it to die. So IMO removing this means we should have some other way. Maybe the middleManager should heartbeat to its tasks, and they should exit if they haven't got a heartbeat in a certain amount of time? Something long enough to cover any reasonable update + restart time.